plugins/frontend/backstage-plugin-prometheus/src/hooks/usePrometheus.ts

```diff
@@ -132,10 +132,23 @@ export function useAlerts(alerts: string[] | 'all') {
     }, []);
   if (value && value.status !== 'error') {
     const rules = value.data.groups.flatMap(it => it.rules);
+    const labels = getLabels(entity);
+    const parsedLabels = labels
+      ? Object.fromEntries(labels.split(',').map(label => label.split('=')))
+      : undefined;
     const displayableAlerts: PrometheusDisplayableAlert[] = rules
       .filter(rule => alerts === 'all' || alerts.includes(rule.name))
       .flatMap(rule =>
         rule.alerts.map(alert => ({ ...rule, ...alert, id: rule.name })),
+      )
+      .filter(alert =>
+        parsedLabels
+          ? Object.keys(parsedLabels).every(
+              key =>
+                alert.labels[key as keyof typeof alert.labels] ===
```

Comment:
can `key as keyof typeof alert.labels` ever be anything else than `string`?

Reply:
I don't think so, but the reason I have used to awful expression was because of this error:
```
Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ alertname: string; }'.
```
This expression seems to work because it does cast key as different type than string. Other option I was thinking could work would be to create interface and cast labels as that interface but I wanted to change smallest number of things in original code so I went with this solution WDYT?

Reply:
Ok.. so `alert.labels[key as string]` doesn't work?

Reply:
It will work but eslint will fail with beforementioned error which I guess we could disable eslint for this line but I don't know if that is a good idea...